### ISSUE TITLE
Support for binary requests in persister

### ISSUE THREAD

**Issue by therve** [Netflix/pollyjs#442]
## Description

Currently when we send binary requests (in our cases, JSON compressed payloads), the persister doesn't record anything. The core of the issue is that HAR doesn't really support it, given that it only has a `text` property in `postData`. I believe it can be worked around by using hex/base64, and find a way to mark the payload binary (maybe just add a _isBinary field? Or additional data in the mime type?).

## Relevant Links

https://github.com/Netflix/pollyjs/blob/18262cf3213d3f3ca7a778a574c162dc2318860a/packages/%40pollyjs/persister/src/har/request.js#L43

**Comment by offirgolan** [comment]
As of right now, we don't support it mostly due to the fact that we don't need the post data when replaying recordings. Is there a specific use case you have for needing this?

**Comment by therve** [comment]
Yeah I want my test run to fail if the payload has changed and there is no recording matching. I can do that with JSON payloads today, but not compressed payloads.

**Comment by offirgolan** [comment]
Would setting [recordIfMissing](https://netflix.github.io/pollyjs/#/configuration?id=recordifmissing) config to `false` solve this?
